service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/cmo/ReviewCMOController.java

```diff
@@ -0,0 +1,156 @@
+package uk.gov.hmcts.reform.fpl.controllers.cmo;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.swagger.annotations.Api;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.events.CaseManagementOrderIssuedEvent;
+import uk.gov.hmcts.reform.fpl.events.CaseManagementOrderRejectedEvent;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.ReviewDecision;
+import uk.gov.hmcts.reform.fpl.model.common.DocumentReference;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.common.dynamic.DynamicList;
+import uk.gov.hmcts.reform.fpl.model.order.CaseManagementOrder;
+import uk.gov.hmcts.reform.fpl.service.DocumentSealingService;
+import uk.gov.hmcts.reform.fpl.service.cmo.ReviewCMOService;
+import uk.gov.hmcts.reform.fpl.service.docmosis.DocumentConversionService;
+
+import java.util.List;
+import java.util.Map;
+
+import static uk.gov.hmcts.reform.fpl.enums.CMOReviewOutcome.JUDGE_REQUESTED_CHANGES;
+import static uk.gov.hmcts.reform.fpl.enums.CMOStatus.RETURNED;
+
+@Api
+@RestController
+@RequestMapping("/callback/review-cmo")
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class ReviewCMOController {
+
+    private final ObjectMapper mapper;
+    private final ReviewCMOService reviewCMOService;
+    private final DocumentConversionService documentConversionService;
+    private final DocumentSealingService documentSealingService;
+    private final ApplicationEventPublisher eventPublisher;
+
+    @PostMapping("/about-to-start")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        Map<String, Object> data = caseDetails.getData();
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        List<Element<CaseManagementOrder>> cmosReadyForApproval = reviewCMOService.getCMOsReadyForApproval(
+            caseData.getDraftUploadedCMOs());
+
+        data.remove("reviewCMODecision");
+
+        data.putAll(reviewCMOService.handlePageDisplayLogic(cmosReadyForApproval));
```

Comment:
reviewCMOService.handlePageDisplayLogic should accept CaseData, controller will simplify